lua/dap/ui/variables.lua

```diff
@@ -99,6 +107,34 @@ local function update_variable_buffer(buf, win)
 end
 
 
+function M.toggle_data_breakpoint(condition, hit_condition)
+  local buf = api.nvim_get_current_buf()
+  local state = variable_buffers[buf]
+  local pos = api.nvim_win_get_cursor(0)
+  if not state then return end
+
+  local variable = state.line_to_variable[pos[1] - 1]
+
+  if variable then
+    local parent
+    for _, p in pairs(state.line_to_variable) do
```

Comment:
this will only work when the parent is also display in this variable buffer. So if the breakpoint is set on any of the non-root elements. A depth search on `current_frame.variables` might be more precise.